hgshah/cmput404-project
# Call remote authors

# Overview

Make local node make remote calls on your behalf.

## Design decision

To indicate we want to do things outside our local node. We add a node parameter, which should be equal to the domain of that server.

This is based on how Mastodon does it (partially) where they append the url with the remote server the account was originally from, like `https://mastodon.yourserver/@account@differentserver`

Ours follow: `http://service/endpoint/?node=domain.herokuapp.com`

## Changes
- Add remote logic for Author endpoint
- Add RemoteConfigs which has overrideable values and logic that can be on a per server (per python class) basis

## Testing
Go to the following urls
- See all local authors: https://potato-oomfie.herokuapp.com/authors/
- See all remote authors: https://potato-oomfie.herokuapp.com/authors/?node=turnip-oomfie-1.herokuapp.com
- See a local author: http://turnip-oomfie-1.herokuapp.com/authors/656821bc-6ca1-4fe3-ba1b-927e9860d0a4/
- See a remote author: http://potato-oomfie.herokuapp.com/authors/656821bc-6ca1-4fe3-ba1b-927e9860d0a4/?node=turnip-oomfie-1.herokuapp.com
- Pagination: https://turnip-oomfie-1.herokuapp.com/authors/?node=potato-oomfie.herokuapp.com&page=2&size=1
